src/System.Configuration/src/System/Configuration/TypeUtil.cs

```diff
@@ -9,25 +9,65 @@ namespace System.Configuration
 {
     internal static class TypeUtil
     {
-        // Since the config APIs were originally implemented in System.dll,
-        // references to types without assembly names could be resolved to
-        // System.dll in Everett. Emulate that behavior by trying to get the
-        // type from System.dll
-        private static Type GetLegacyType(string typeString)
+        // Deliberately not being explicit about the versions to make
+        // things simpler for consumers of System.Configuration.
+        private static string[] s_implicitAssemblies =
         {
-            Type type = null;
+            "System.Runtime",
+            "System.Runtime.Extensions",
+            "System.Collections",
+            "System.Collections.Concurrent",
+            "System.Collections.Specialized",
+
+            // Historically this is the assembly we would look in
```

Comment:
Let's say, this code used to live in System on desktop, so an unqualified type name would have looked first in System then in mscorlib.

Given that, I think this list should start with "System", then list "mscorlib".

All this is based on some cursory tests of a desktop console app.  We should probably take a close look at what desktop did and make sure that this is a correct model.

Now that we're bringing back all the classic assemblies as a *part of the framework* I wonder if we could have reflection do the mscorlib check.  @weshaggard what do you think?